catppuccin.tmux

```diff
@@ -31,6 +31,160 @@ setw() {
+    local show_number="#[fg=$background,bg=$color]$number"
+    local show_middle_separator="#[fg=$background,bg=$color,nobold,nounderscore,noitalics]$window_middle_separator"
+    local show_text="#[fg=$background,bg=$color]$text"
+    local show_right_separator="#[fg=$color,bg=$thm_bg]$window_right_separator"
+
+  fi
+  
+  if [[ $fill == "number" ]] 
+  then
+    local show_number="#[fg=$background,bg=$color]$number"
+    local show_middle_separator="#[fg=$color,bg=$background,nobold,nounderscore,noitalics]$window_middle_separator"
+    local show_text="#[fg=$thm_fg,bg=$background]$text"
+
+    if [[ $window_number_position == "right" ]]
+    then
+      local show_left_separator="#[fg=$background,bg=$thm_bg,nobold,nounderscore,noitalics]$window_left_separator"
+      local show_right_separator="#[fg=$color,bg=$thm_bg]$window_right_separator"
+    fi
+
+    if [[ $window_number_position == "left" ]]
+    then
+      local show_right_separator="#[fg=$background,bg=$thm_bg,nobold,nounderscore,noitalics]$window_right_separator"
+      local show_left_separator="#[fg=$color,bg=$thm_bg]$window_left_separator"
+    fi
+
+  fi
+
+  local window_format
+
+  if [[ $window_number_position == "right" ]]
+  then
+    window_format="$show_left_separator$show_text$show_middle_separator$show_number$show_right_separator"
+  fi
+
+  if [[ $window_number_position == "left" ]]
+  then
+    window_format="$show_left_separator$show_number$show_middle_separator$show_text$show_right_separator"
+  fi
+
+  echo $window_format
+}
+
+build_status_module() {
+  local index="$1"
+  local icon="$2"
+  local color="$3"
+  local text="$4"
+
+  if [[ $index -eq 0 || $status_connect_separator == "no" ]]
+  then
+    local show_left_separator="#[fg=$color,bg=$thm_bg,nobold,nounderscore,noitalics]$status_left_separator"
+  else
+    local show_left_separator="#[fg=$color,bg=$thm_gray,nobold,nounderscore,noitalics]$status_left_separator"
+  fi
+
+  if [[ $status_fill == "icon" ]]
+  then
+    local show_icon="#[fg=$thm_bg,bg=$color,nobold,nounderscore,noitalics]$icon "
+    local show_text="#[fg=$thm_fg,bg=$thm_gray] $text"
+    local show_right_separator="#[fg=$thm_gray,bg=$thm_bg,nobold,nounderscore,noitalics]$status_right_separator"
+  fi
+
+  if [[ $status_fill == "all" ]]
+  then
+    local show_icon="#[fg=$thm_bg,bg=$color,nobold,nounderscore,noitalics]$icon "
+    local show_text="#[fg=$thm_bg,bg=$color]$text"
+    local show_right_separator="#[fg=$color,bg=$thm_bg,nobold,nounderscore,noitalics]$status_right_separator"
+  fi
+
+  if [[ $status_right_separator_inverse == "yes" ]]
+  then
+    local show_right_separator="#[fg=$thm_bg,bg=$color,nobold,nounderscore,noitalics]$status_right_separator"
+  fi
+
+  echo "$show_left_separator$show_icon$show_text$show_right_separator"
+}
+
+load_modules() {
+  local loaded_modules
+
+  local modules_path=$1
+  local modules_list=$2
+
+  local modules_array
+  read -a modules_array <<< "$modules_list"
```

Comment:
Arrays and `<<<` are Bashisms, the portable equivalent would be to remove this variable and replace the for loop below with `echo "$modules_list" | while read -r module_name`.

Reply:
@Stonks3141 I am scratching my head here trying to implement this suggestion. The problem is that i need to update 2 variables from inside of the loop. From my understanding using a "|" creates a subshell where the while loop runs and the variables outside of the subshell will not get updated.

Any ideas on how to move forward with this?